IRremoteESP8266.cpp

```diff
@@ -427,24 +427,76 @@ void ICACHE_FLASH_ATTR IRsend::sendRCMM(uint32_t data, uint8_t nbits) {
 
 }
 
+// Send a Panasonic formatted message.
+//
+// Args:
+//   data:   The message to be sent.
+//   nbits:  The number of bits of the message to be sent. (PANASONIC_BITS).
+//   repeat: The number of times the command is to be repeated.
+//
+// Note:
+//   This protocol is a modified version of Kaseikyo.
+void ICACHE_FLASH_ATTR IRsend::sendPanasonic64(unsigned long long data,
+                                               unsigned int nbits,
+                                               unsigned int repeat) {
+  enableIROut(37);  // Set IR carrier frequency
+  IRtimer usecTimer = IRtimer();
+
+  for (uint16_t i = 0; i <= repeat; i++) {
+    usecTimer.reset();
+    // Header
+    mark(PANASONIC_HDR_MARK);
+    space(PANASONIC_HDR_SPACE);
+    // Data
+    sendData(PANASONIC_BIT_MARK, PANASONIC_ONE_SPACE,
+             PANASONIC_BIT_MARK, PANASONIC_ZERO_SPACE,
+             data, nbits, true);
+    // Footer
+    mark(PANASONIC_BIT_MARK);
+    space(max(PANASONIC_MIN_COMMAND_LENGTH - usecTimer.elapsed(),
+              PANASONIC_MIN_GAP));
+  }
+}
+
+// Send a Panasonic formatted message.
+//
+// Args:
+//   address: The manufacturer code.
+//   data:    The data portion to be sent.
+//   nbits:   The number of bits of the message to be sent. (PANASONIC_BITS).
+//   repeat:  The number of times the command is to be repeated.
+//
+// Note:
+//   This protocol is a modified version of Kaseikyo.
 void ICACHE_FLASH_ATTR IRsend::sendPanasonic(unsigned int address,
-                                             unsigned long data) {
-  // Set IR carrier frequency
-  enableIROut(35);
-  // Header
-  mark(PANASONIC_HDR_MARK);
-  space(PANASONIC_HDR_SPACE);
-  // Address (16 bits)
-  sendData(PANASONIC_BIT_MARK, PANASONIC_ONE_SPACE,
-           PANASONIC_BIT_MARK, PANASONIC_ZERO_SPACE,
-           address, 16, true);
-  // Data (32 bits)
-  sendData(PANASONIC_BIT_MARK, PANASONIC_ONE_SPACE,
-           PANASONIC_BIT_MARK, PANASONIC_ZERO_SPACE,
-           data, 32, true);
-  // Footer
-  mark(PANASONIC_BIT_MARK);
-  ledOff();
+                                             unsigned long data,
+                                             unsigned int nbits,
+                                             unsigned int repeat) {
```

Comment:
Yes, we are. The previous frequency was incorrect (35kHz), all the references I could find, and I found a lot, point to it being 37kHz. I'll be creating a PR for master trunk soon to fix it there.

My experience has been most devices are tolerant to with in a few kHz. If it was working at 35 for people when it should have been 37, it kind of backs that up. Also see #36 which indicates things are not all rosy in the Panasonic world.